br-external/package/linux_firmware_next/linux_firmware_next.sh

```diff
@@ -0,0 +1,155 @@
+        echo "tar cJf "$PATH_TO_DL_LINUX_FIRMWARE/linux-firmware-next.tar.xz $PATH_TO_LINUX_FIRMWARE_NEXT "(this operation may take a while)"
+        tar -C $PATH_TO_LINUX_FIRMWARE_NEXT_PARENT  --transform="s/$LINUX_FIRMWARE_NEXT_FILENAME/$LINUX_FIRMWARE_NEXT_PROPER_FILENAME/" -cJf $PATH_TO_TAR $LINUX_FIRMWARE_NEXT_FILENAME --checkpoint=.100
+    fi
+}
+
+# Update makefile linux_firmware.mk to change version (on version with linux firmware next)
+update_makefile()
+{
+    line=$(grep 'LINUX_FIRMWARE_VERSION =' package/linux-firmware/linux-firmware.mk)
+    output="LINUX_FIRMWARE_VERSION = next"
+
+    sed -i "s|$line|$output|" package/linux-firmware/linux-firmware.mk
+}
+
+# Update hash file (linux-firmware.hash):
+#    arg $1: file based on which the hash is calculated
+#    arg $2: string in the hash file that should be replaced 
+update_hash_file()
+{
+    line=$(grep $2 $PATH_TO_HASH)
+    sha256="$(sha256sum $1 | cut -d' ' -f1)"
+    filename=$(basename $1)
+    echo "sha256 " $sha256 " " $filename
+
+    output="sha256 "$sha256" "$filename
+    sed -i "s|$line|$output|" $PATH_TO_HASH  
+}
+
+update_hash_tar()
+{
+    update_hash_file $PATH_TO_TAR "linux-firmware-"
+}
+
+update_hash_license()
+{
+    tab=(
+        "LICENCE.Abilis"
+        "LICENSE.amdgpu"
+        "LICENCE.Marvell"
+        "LICENCE.atheros_firmware"
+        "ath10k/QCA6174/hw3.0/notice_ath10k_firmware-4.txt"
+        "ath10k/QCA6174/hw2.1/notice_ath10k_firmware-5.txt"
+        "ath10k/QCA6174/hw3.0/notice_ath10k_firmware-6.txt"
+        "LICENCE.broadcom_bcm43xx"
+        "LICENCE.chelsio_firmware"
+        "LICENCE.cypress"
+        "LICENCE.fw_sst_0f28"
+        "LICENCE.ibt_firmware"
+        "LICENSE.ice_enhanced"
+        "LICENCE.it913x"
+        "LICENCE.iwlwifi_firmware"
+        "LICENCE.microchip"
+        "LICENCE.moxa"
+        "LICENCE.qat_firmware"
+        "LICENCE.qla2xxx"
+        "LICENCE.ralink-firmware.txt"
+        "LICENCE.ralink_a_mediatek_company_firmware"
+        "LICENCE.rtlwifi_firmware.txt"
+        "LICENCE.ti-connectivity"
+        "LICENCE.xc4000"
+        "LICENCE.xc5000"
+        "LICENCE.xc5000c"
+        "LICENSE.QualcommAtheros_ar3k"
+        "LICENSE.QualcommAtheros_ath10k"
+        "LICENSE.dib0700"
+        "LICENSE.i915"
+        "LICENSE.qcom"
+        "LICENSE.radeon"
+        "LICENSE.sdma_firmware"
+        "WHENCE"
+        "qcom/NOTICE.txt"
+        "LICENCE.e100"
+    );
+
+    for element in "${tab[@]}"; do
+        update_hash_file $PATH_TO_LINUX_FIRMWARE_NEXT"/"$element $element
+    done
+}
+
+update_hash()
+{
+    update_hash_tar
+    update_hash_license
+}
+
+# Filesystem based on linux-firmware-next (rootfs.cpio.gz) is to big to fit in disk.img (100MB). Size od disk.img must be increased (512MB) 
+update_size_of_img()
+{
+    line=$(grep 'efi_part_size=' $PATH_TO_POST_IMAGE_EFI_GPT)
+    output='efi_part_size=$(( 536870912 / 512 )) # 512MB'
+
+    sed -i "s|$line|$output|" $PATH_TO_POST_IMAGE_EFI_GPT
```

Comment:
Add a check to see if sed is necessary in case of identical lines.

Reply:
Done in commit edc0ed00b3e35507dce1b0195b35db3b4fa1d796